core/src/authentication.rs

```diff
@@ -96,7 +96,9 @@ impl Credentials {
                 return Err(AuthenticationError::Key.into());
             }
 
-            pbkdf2::<Hmac<Sha1>>(&secret, username.as_bytes(), 0x100, &mut key[0..20]);
+            let res = pbkdf2::<Hmac<Sha1>>(&secret, username.as_bytes(), 0x100, &mut key[0..20]);
+            assert!(res.is_ok(),"Failed to create hash");
```

Comment:
I think you should go for the [`pbkdf2_hmac`](https://docs.rs/pbkdf2/0.12.1/pbkdf2/fn.pbkdf2_hmac.html) function instead, since that wraps this function and handles the result properly.

Reply:
Thanks, changed it.